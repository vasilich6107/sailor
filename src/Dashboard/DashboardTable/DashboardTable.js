// @flow
import React from 'react';
import DashboardColumn from '../DashboardColumn/DashboardColumn';
import './DashboardTable.css';
import type { DashboardColumnsType } from '../../types/Dashboard';
import type { ApplicantType } from '../../types/Applicant';

type Props = {
  columnOrder: DashboardColumnsType,
  applicants: { [key: string]: Array<ApplicantType> },
  onStateChange: (newState: string) => (uuid: string) => void,
};

const DashboardTable = ({ columnOrder, applicants, onStateChange }: Props) => (
  <div className="dashboard">
    {columnOrder.map((column: string, index: number, source: DashboardColumnsType) => {
      const nextState: string = source[index + 1];
      const prevState: string = source[index - 1];

      return (
        <DashboardColumn
          key={column}
          columnName={column}
          applicants={applicants[column]}
          onNext={nextState && onStateChange(nextState)}
          onPrevious={prevState && onStateChange(prevState)}
        />
      );
    })}
  </div>
);

export default DashboardTable;
