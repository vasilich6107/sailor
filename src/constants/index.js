export const APPLICANT_STATES = {
  APPLIED: 'APPLIED',
  INTERVIEWING: 'INTERVIEWING',
  HIRED: 'HIRED',
};
