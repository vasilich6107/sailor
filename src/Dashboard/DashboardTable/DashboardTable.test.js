import React from 'react';
import { shallow } from 'enzyme';
import DashboardTable from './DashboardTable';
import { APPLICANT_STATES } from '../../constants';

jest.mock('../DashboardColumn', () => 'DashboardColumn');

const applicants = {
  [APPLICANT_STATES.APPLIED]: [
    {
      name: {
        first: 'francis',
        last: 'stewart',
        title: 'mr',
      },
    },
  ],
  [APPLICANT_STATES.INTERVIEWING]: [
    {
      name: {
        first: 'harvey',
        last: 'clarke',
        title: 'mr',
      },
    },
  ],
  [APPLICANT_STATES.HIRED]: [
    {
      name: {
        first: 'mia',
        last: 'white',
        title: 'mr',
      },
    },
  ],
};

export const DASHBOARD_COLUMNS = [APPLICANT_STATES.APPLIED, APPLICANT_STATES.INTERVIEWING, APPLICANT_STATES.HIRED];

it('Renders dashboard column', () => {
  const handleStateChange = jest.fn().mockImplementation(() => () => {});

  expect(
    shallow(
      <DashboardTable columnOrder={DASHBOARD_COLUMNS} applicants={applicants} onStateChange={handleStateChange} />
    )
  ).toMatchSnapshot();

  expect(handleStateChange.mock.calls.length).toBe(4);
  expect(handleStateChange.mock.calls[0][0]).toBe(APPLICANT_STATES.INTERVIEWING);
  expect(handleStateChange.mock.calls[1][0]).toBe(APPLICANT_STATES.HIRED);
  expect(handleStateChange.mock.calls[2][0]).toBe(APPLICANT_STATES.APPLIED);
  expect(handleStateChange.mock.calls[3][0]).toBe(APPLICANT_STATES.INTERVIEWING);
});
