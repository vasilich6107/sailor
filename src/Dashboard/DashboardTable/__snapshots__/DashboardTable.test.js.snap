// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Renders dashboard column 1`] = `
<div
  className="dashboard"
>
  <DashboardColumn
    applicants={
      Array [
        Object {
          "name": Object {
            "first": "francis",
            "last": "stewart",
            "title": "mr",
          },
        },
      ]
    }
    columnName="APPLIED"
    key="APPLIED"
    onNext={[Function]}
  />
  <DashboardColumn
    applicants={
      Array [
        Object {
          "name": Object {
            "first": "harvey",
            "last": "clarke",
            "title": "mr",
          },
        },
      ]
    }
    columnName="INTERVIEWING"
    key="INTERVIEWING"
    onNext={[Function]}
    onPrevious={[Function]}
  />
  <DashboardColumn
    applicants={
      Array [
        Object {
          "name": Object {
            "first": "mia",
            "last": "white",
            "title": "mr",
          },
        },
      ]
    }
    columnName="HIRED"
    key="HIRED"
    onPrevious={[Function]}
  />
</div>
`;
